docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,329 @@
+passed to `DAGStore`.
+
+NOTE: Registering on the DAGStore automatically populates top-level index with shares/proofs accessible from stored EDS
+so this is out of scope of the document.
+
+```go
+// Put stores the given data square with DataRoot as key.
+//
+// The square is verified on the Exchange level and Put only stores the square trusting it.
+// The resulting file stores all the shares and NMT Merkle Proofs of the EDS.
+// Additionally, the file gets indexed s.t. Store.Blockstore can access them. 
+func (s *Store) Put(context.Context, DataRoot, *rsmt2d.ExtendedDataSquare) error
+```
+
+##### `share.EDSStore.GetCAR`
+
+To read an EDS as a byte stream `GetCAR` method is introduced. Internally it
+
+- Converts `DataRoot` into the [`shard.Key`](<https://github.com/filecoin-project/dagstore/blob/master/shard/key.go#L12>
+- Gets Mount by `shard.Key` from [`mount.Registry`](https://github.com/filecoin-project/dagstore/blob/master/mount/registry.go#L22)
+- Returns `io.ReadeCloser` from [`Mount.Fetch`](https://github.com/filecoin-project/dagstore/blob/master/mount/mount.go#L71)
+
+NOTE: It might be necessary to acquire EDS mount via `DAGStore` keeping `ShardAccessor`
+and closing it when operation is done. This has to be confirmed.
+
+```go
+// GetCAR takes DataRoot and returns a buffered reader to respective EDS serialized as CARv1 file.
+// 
+// The Reader strictly reads the first quadrant(1/4) of EDS omitting all the NMT Merkle proofs.
+// Integrity of the store data is not verified. 
+// 
+// Caller must Close returned reader after reading.
+func (s *Store) GetCAR(context.Context, DataRoot) (io.ReadCloser, error)
+```
+
+##### `share.EDSStore.Blockstore`
+
+`Blockstore` method return a `Blockstore` interface implementation instance, providing random access over share and NMT
+Merkle proof in every stored EDS. It is required for FNs/BNs to serve DAS requests over the Bitswap and for reading data
+by namespace. // TODO Link to the section
+
+There is a [frozen/un-merged implementation](https://github.com/filecoin-project/dagstore/pull/116) of `Blockstore`
+over `DAGStore` and CARv2 indexes.
+
+NOTE: We can either use it(and finish if something is missing for our case) or implement custom optimized for our needs.
+
+```go
+// Blockstore returns an IPFS Blockstore providing access to individual shares/nodes of all EDS 
+// registered on the Store. NOTE: The Blockstore does not store full Celestia Blocks, but IPFS blocks. 
+// We represent `shares` and NMT Merkle proofs as IPFS blocks and IPLD nodes, so that Bitswap can access those.
+func (s *Store) Blockstore() blockstore.Blockstore
+
+```
+
+##### `share.EDSStore.Get`
+
+To read an entire EDS `Get` method is introduced. Internally it:
+
+- Gets serialized EDS `io.Reader` via `Store.GetCAR`
+- Deserializes EDS and validates it via `share.ReadEDS`
+
+NOTE: It's not necessary, but an API ergonomics/symmetry nice-to-have
```

Comment:
I do think this is necessary! Not just nice-to-have :) Something would very much be missing without it

Reply:
I mean for the networking protocol and its not required and it would use just `GetCAR`

Reply:
(this is coming from Ismail always wanting me in the past to be pragmatic and implement what is absolutely required 😁)